### PR TITLE
Fix backwards compatibility for tx fee

### PR DESCRIPTION
**Pull request by SebastianMarian**
* Fixed backwards compatibility for calculation of tx fee in "checkTxValues" method from baseTxProcessor and "Fee" method from InterceptedTransaction

### CODE REVIEW

File: cmd/node/main.go [hunk 1]
```diff
@@ -833,7 +833,11 @@ func startNode(ctx *cli.Context, log logger.Logger, version string) error {
 	time.Sleep(secondsToWaitForP2PBootstrap * time.Second)
 
 	log.Trace("creating economics data components")
-	economicsData, err := economics.NewEconomicsData(economicsConfig)
+	economicsData, err := economics.NewEconomicsData(
+		economicsConfig,
```
Comment (sasurobert) [inline cmd/node/main.go:837]: add arguments structure instead of list of arguments

File: process/economics/economicsData.go [hunk 2]
```diff
@@ -198,10 +215,16 @@ func (ed *EconomicsData) GasPerDataByte() uint64 {
...
 // ComputeMoveBalanceFee computes the provided transaction's fee
 func (ed *EconomicsData) ComputeMoveBalanceFee(tx process.TransactionWithFeeHandler) *big.Int {
-	gasPrice := big.NewInt(0).SetUint64(tx.GetGasPrice())
-	gasLimit := big.NewInt(0).SetUint64(ed.ComputeGasLimit(tx))
+	return core.SafeMul(tx.GetGasPrice(), ed.ComputeGasLimit(tx))
+}
 
-	return gasPrice.Mul(gasPrice, gasLimit)
+// EstimateMoveBalanceFee estimates the provided transaction's fee
+func (ed *EconomicsData) EstimateMoveBalanceFee(tx process.TransactionWithFeeHandler) *big.Int {
```
Comment (sasurobert) [inline process/economics/economicsData.go:222]: Wrong function name - we will get into the same problem later on.

File: process/transaction/baseProcess.go [hunk 3]
```diff
@@ -137,7 +137,7 @@ func (txProc *baseTxProcessor) checkTxValues(
 		return process.ErrWrongTypeAssertion
 	}
 
-	txFee := core.SafeMul(tx.GasLimit, tx.GasPrice)
+	txFee := txProc.economicsFee.EstimateMoveBalanceFee(tx)
```
Comment (sasurobert) [inline process/transaction/baseProcess.go:140]: ComputeTxFee

**Review by sasurobert — CHANGES_REQUESTED** [review]

**Review by sasurobert — APPROVED** [review]

File: cmd/node/main.go (hunk 1 above)
Comment (SebastianMarian) [reply cmd/node/main.go]: done

File: process/economics/economicsData.go (hunk 2 above)
Comment (SebastianMarian) [reply process/economics/economicsData.go]: done

File: process/transaction/baseProcess.go (hunk 3 above)
Comment (SebastianMarian) [reply process/transaction/baseProcess.go]: done

**Review by AdoAdoAdo — APPROVED** [review]

**Review by LucianMincu — APPROVED** [review]
System tests passed.

**Pull request merged**